src/query/parallel/px_hash_join.cpp

```diff
@@ -0,0 +1,985 @@
+      HASHJOIN_START_STATS start_stats = HASHJOIN_START_STATS_INITIALIZER;
+#if HASHJOIN_PROFILE_TIME
+      HASHJOIN_START_STATS profile_start_stats = HASHJOIN_START_STATS_INITIALIZER;
+#endif /* HASHJOIN_PROFILE_TIME */
+      assert (!thread_is_on_trace (&thread_ref) || stats != nullptr);
+
+      task_manager task_manager (manager->px_worker_pool_manager->get_worker_pool (),
+				 cuberr::context::get_thread_local_context ());
+      join_task *task = nullptr;
+
+      if (thread_is_on_trace (&thread_ref))
+	{
+	  stats->build.min_elapsed_time = { LONG_MAX, 999999 };
+	  stats->probe.min_elapsed_time = { LONG_MAX, 999999 };
+
+	  hjoin_trace_start (&thread_ref, &start_stats);
+	}
+
+      for (context_index = 0; context_index < manager->context_cnt; context_index++)
+	{
+	  current_context = &manager->contexts[context_index];
+
+	  task = new join_task (task_manager, manager, current_context);
+	  task_manager.push_task (task);
+	}
+
+      task_manager.join ();
+
+      if (thread_is_on_trace (&thread_ref))
+	{
+	  hjoin_trace_end (&thread_ref, &stats->parallel, &start_stats);
+	}
+
+      if (task_manager.has_error ())
+	{
+	  assert_release (er_errid () != NO_ERROR);
+	  return er_errid ();
+	}
+
+      for (context_index = 0; context_index < manager->context_cnt; context_index++)
+	{
+	  current_context = &manager->contexts[context_index];
+
+	  if (thread_is_on_trace (&thread_ref))
+	    {
+	      hjoin_trace_merge_stats (stats, current_context->stats);
+	    }
+
+	  if (current_context->list_id == nullptr)
+	    {
+	      error = er_errid ();
+	      if (error != NO_ERROR)
+		{
+		  assert_release (er_errid () != NO_ERROR);
+		  return er_errid ();
+		}
+	      else
+		{
+		  /* list_id can be NULL when the join result is empty.
+		   * In this case, it is NO_ERROR. */
+		  continue;
+		}
+	    }
+
+	  if (current_context->list_id->tuple_cnt == 0)
+	    {
+	      qfile_destroy_list (&thread_ref, current_context->list_id);
+	      QFILE_FREE_AND_INIT_LIST_ID (current_context->list_id);
+
+	      /* empty context */
+	      continue;
+	    }
+
+	  HJOIN_PROFILE_START (&thread_ref, &profile_start_stats, HASHJOIN_PROFILE_MERGE);
+	  error = hjoin_merge_qlist (&thread_ref, manager, current_context);
+	  HJOIN_PROFILE_MERGE_END (&thread_ref, &stats->profile, &profile_start_stats, HASHJOIN_PROFILE_MERGE,
+				   manager->single_context.list_id->tuple_cnt);
+
+	  if (error != NO_ERROR)
+	    {
+	      assert_release (er_errid () != NO_ERROR);
+	      return er_errid ();
+	    }
+	}
+
+      ASSERT_NO_ERROR_OR_INTERRUPTED ();
+      return NO_ERROR;
+    }
+
+    /*
+     * tls_spawner
+     */
+
+    thread_local std::unique_ptr<cubxasl::spawner> tls_spawner;
+    thread_local VAL_DESCR *tls_val_descr = nullptr;
+    thread_local PRED_EXPR *tls_during_join_pred = nullptr;
+    thread_local REGU_VARIABLE_LIST tls_outer_regu_list_pred = nullptr;
+    thread_local REGU_VARIABLE_LIST tls_inner_regu_list_pred = nullptr;
```

Comment:
thread_local 정보들에 대해서 생성자/소멸자 및 접근/할당/해제가 명확하게 관리되게끔 class로 관리되었으면 좋겠습니다. 예를 들면 다음과 같이 작성하면 어떨까요?
```suggestion
    class thread_local_storage
    {
      std::unique_ptr<cubxasl::spawner> spawner;
      VAL_DESCR *val_descr;
      PRED_EXPR *during_join_pred;
      REGU_VARIABLE_LIST outer_regu_list_pred;
      REGU_VARIABLE_LIST inner_regu_list_pred;
    };
    thread_local thread_local_storage *tls_storage;
```

Reply:
좀 더 고민한 후에 변경 여부를 말씀드리겠습니다.

Reply:
반영했습니다. ([6f3e74a](https://github.com/CUBRID/cubrid/pull/6247/commits/6f3e74a60ec3a996e6b74b8aead991b03d5587cf))